code/modules/mob/living/silicon/robot/life.dm

```diff
@@ -157,20 +90,6 @@
 	else
 		throw_alert("charge", /obj/screen/alert/nocell)
 
-/mob/living/silicon/robot/proc/update_items()
-	if (client)
-		client.screen -= contents
-		for(var/obj/I in contents)
-			if(I && !(istype(I,/obj/item/weapon/stock_parts/cell) || istype(I,/obj/item/device/radio)  || istype(I,/obj/machinery/camera) || istype(I,/obj/item/device/mmi)))
-				client.screen += I
-	if(module_state_1)
-		module_state_1:screen_loc = ui_inv1
-	if(module_state_2)
-		module_state_2:screen_loc = ui_inv2
-	if(module_state_3)
-		module_state_3:screen_loc = ui_inv3

```

Comment:
we no longer call this proc every life(). these hud updates are now done in code/on_click/hud/robot.